File: python/native_get_latest_point.py
from client import create_gateway_client
import compassiot.native.v1.native_pb2 as native


def main():
    client = create_gateway_client()

    request = native.GetLatestPointRequest(
        vin=""
    )

    response = client.NativeGetLatestPoint(request)

    print(response.event)

     
if __name__ == "__main__":
    main()

File: typescript/src/native_get_latest_point.ts
import * as native from "@buf/compassiot_api.bufbuild_es/compassiot/native/v1/native_pb"

import { createNodeClient } from "./client"

const client = createNodeClient()

const request = new native.GetLatestPointRequest({
  vin: ""
})

const response = await client.nativeGetLatestPoint(request)

console.log(response.toJsonString({ prettySpaces: 2 }))
